contracts/instruments/RibbonThetaVaultYearn.sol

```diff
@@ -598,6 +608,45 @@ contract RibbonThetaVaultYearn is DSMath, OptionsVaultStorage {
         emit CapSet(oldCap, newCap, msg.sender);
     }
 
+    /**
+     * @notice Closes the vault and makes it withdraw only.
+     */
+    function sunset(address upgradeTo) external onlyOwner {
+        require(upgradeTo != address(0), "!upgradeTo");
+
+        replacementVault = IRibbonV2Vault(upgradeTo);
+
+        emit VaultSunset(upgradeTo);
+    }
+
+    /*
+     * @notice Moves msg.sender's deposited funds to new vault w/o fees
+     */
+    function migrate() external nonReentrant {
+        IRibbonV2Vault vault = replacementVault;
+        require(address(vault) != address(0), "Not sunset");
+
+        uint256 maxShares = maxWithdrawableShares();
+        uint256 share = balanceOf(msg.sender);
+        uint256 allShares = min(maxShares, share);
+
+        (uint256 withdrawAmount, uint256 feeAmount) =
+            withdrawAmountWithShares(allShares);
+
+        // Since we want to exclude fees, we add them both together
+        withdrawAmount = withdrawAmount.add(feeAmount);
+
+        emit Migrate(msg.sender, address(vault), allShares, withdrawAmount);
+
+        _burn(msg.sender, allShares);
+
+        _unwrapYieldToken(withdrawAmount);
```

Comment:
assuming you know that there is no need for _unwrapYieldToken because we can always do depositYieldToken(). so guessing its because of yvusdc version incompatibility

Reply:
Indeed, that is the case here. Just to clarify:
- v1 vault uses yvUSDC 0.3.0
- v2 vault uses yvUSDC v0.4.3